modules/mauth-authenticator-akka-http/src/main/scala/com/mdsol/mauth/akka/http/MAuthDirectives.scala

```diff
@@ -88,16 +107,33 @@ trait MAuthDirectives extends StrictLogging {
       }
     }
 
+  /**
+    * Extracts the detail information of the HTTP request header MCC-Authentication
+    *
+    * @return Directive1[AuthHeaderDetail] of Mauth V2 protocol
+    *         If invalidated, the request is rejected with a MalformedHeaderRejection.
+    */
+  val extractMCCAuthenticationDetail: Directive1[AuthHeaderDetail] =
+    headerValueByType[`MCC-Authentication`]((): Unit).flatMap { hdr =>
+      extractAuthHeaderDetail(hdr.value, MAuthVersion.MWSV2) match {
+        case Some(ahd: AuthHeaderDetail) => provide(ahd)
+        case None =>
+          val msg = s"${`MCC-Authentication`.name} header supplied with bad format: [${hdr.value}]"
+          logger.error(msg)
+          reject(MalformedHeaderRejection(headerName = `MCC-Authentication`.name, errorMsg = msg, None))
+      }
+    }
```

Comment:
I don't think we need these directives. If we delete them we can delete the likes of `MCC-Authentication`. (old values like `extractMwsTimeHeader` will have to be kept for binary compatibility reasons)

Reply:
we should deprecate directives like `extractMwsTimeHeader` too. User should be using `MAuthDirectives#authenticate` to avoid issues and get bug fixes automatically

Reply:
removed directives  of MCC-Authentication, MCC-Time